infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/VcsSshKeysProvisioner.java

```diff
@@ -118,29 +116,39 @@ public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity identity)
       }
     }
 
+    doProvisionSshKey(sshPairs, k8sEnv, identity.getWorkspaceId());
+
     StringBuilder sshConfigData = new StringBuilder();
     for (SshPair sshPair : sshPairs) {
-      doProvisionSshKey(sshPair, k8sEnv, identity.getWorkspaceId());
+
       sshConfigData.append(buildConfig(sshPair.getName()));
     }
 
     String sshConfigMapName = identity.getWorkspaceId() + SSH_CONFIG_MAP_NAME_SUFFIX;
     doProvisionSshConfig(sshConfigMapName, sshConfigData.toString(), k8sEnv);
   }
 
-  private void doProvisionSshKey(SshPair sshPair, KubernetesEnvironment k8sEnv, String wsId) {
-    if (isNullOrEmpty(sshPair.getName()) || isNullOrEmpty(sshPair.getPrivateKey())) {
-      return;
-    }
-    String validNameForSecret = getValidNameForSecret(sshPair.getName());
+  private void doProvisionSshKey(
+      List<SshPairImpl> sshPairs, KubernetesEnvironment k8sEnv, String wsId) {
+
+    Map<String, String> data =
+        sshPairs
+            .parallelStream()
+            .filter(
+                sshPair ->
+                    !(isNullOrEmpty(sshPair.getName()) || isNullOrEmpty(sshPair.getPrivateKey())))
```

Comment:
IMO it's easier to read if the negation is pushed into the condition:
```
!isNullOrEmpty(sshPair.getName()) && !isNullOrEmpty(sshPair.getPrivateKey())
```
since it requires less tracking of parens.

Up to you, however.

Reply:
Updated, thanks!